android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -2720,4 +2723,31 @@ public void onGlobalLayout()
+    public boolean onKeyUp(int keyCode, KeyEvent event) {                                                                                            
+      switch (keyCode) {                                                                                                                             
+        case KeyEvent.KEYCODE_DPAD_DOWN:                                                                                                             
+          Statistics.INSTANCE.trackEvent(Statistics.EventName.ZOOM_OUT);                                                                             
+          AlohaHelper.logClick(AlohaHelper.ZOOM_OUT);                                                                                                
+          MapFragment.nativeScaleMinus();                                                                                                            
+          return true;                                                                                                                               
+        case KeyEvent.KEYCODE_DPAD_UP:                                                                                                               
+          Statistics.INSTANCE.trackEvent(Statistics.EventName.ZOOM_IN);                                                                              
+          AlohaHelper.logClick(AlohaHelper.ZOOM_IN);                                                                                                 
+          MapFragment.nativeScalePlus();                                                                                                             
+          return true;                                                                                                                               
+        case KeyEvent.KEYCODE_ESCAPE:                                                                                                                
+          if (Config.isWunderLINQEnabled()) {                                                                                                        
+            //Go Back To WunderLINQ                                                                                                                  
+            String callingApp = "wunderlinq://datagrid";                                                                                             
+            Intent intent = new Intent(Intent.ACTION_VIEW);                                                                                          
+            intent.setData(Uri.parse(callingApp));                                                                                                   
+            startActivity(intent);                                                                                                                   
```

Comment:
Please reuse the existing method Utils.openUrl(callingApp), since it handles ActivityNotFoundException.

Reply:
When I try this function the app does not launch, this is after removing the http/https check.  I can't seem to see any difference in the way I'm doing it and Utils.openUrl().